zod/cli/torrent_download.py

```diff
@@ -0,0 +1,347 @@
+"""Helper script to download the dataset using academic torrents."""
+
+import os
+import shutil
+import subprocess
+import sys
+import tarfile
+import urllib.request
+from pathlib import Path
+from typing import List
+
+import typer
+from tqdm import tqdm
+
+from zod.cli.download import DownloadSettings, FilterSettings, _print_summary
+from zod.cli.utils import SubDataset, Version
+from zod.constants import MINI
+
+TORRENT_FILEMAP = {
+    SubDataset.FRAMES: "https://academictorrents.com/download/329ae74426e067ef06b82241b5906400ca3caf03.torrent",  # noqa
+    SubDataset.SEQUENCES: "https://academictorrents.com/download/95ece9c22470c4f7df51a82bcc3407cc038419be.torrent",  # noqa
+    SubDataset.DRIVES: "https://academictorrents.com/download/a05ab2e524dc38c1498fcc9fb621329cc97e3837.torrent",  # noqa
+}
+
+SIZE_FILEMAP = {
+    SubDataset.FRAMES: 9096553,
+    SubDataset.SEQUENCES: 1280237,
+    SubDataset.DRIVES: 290441,
+}
+
+DRIVES_FILES_TO_IDX = {
+    "annotations": 1,
+    "drives_mini": 2,
+    "images_front_blur": 3,
+    "infos": 4,
+    "lidar_velodyne": 5,
+    "oxts": 6,
+    "vehicle_data": 7,
+}
+
+SEQUENCES_FILES_TO_IDX = {
+    "annotations": 1,
+    "images_front_blur": 2,
+    "infos": 3,
+    "lidar_velodyne": 4,
+    "oxts": 5,
+    "sequences_mini": 6,
+    "vehicle_data": 7,
+}
+
+FRAMES_FILES_TO_IDX = {
+    "annotations": 1,
+    "frames_mini": 2,
+    "images_front_blur": 3,
+    "images_front_dnat": 4,
+    "infos": 5,
+    "lidar_velodyne_01after": 6,
+    "lidar_velodyne_01before": 7,
+    "lidar_velodyne_02after": 8,
+    "lidar_velodyne_02before": 9,
+    "lidar_velodyne_03after": 10,
+    "lidar_velodyne_03before": 11,
+    "lidar_velodyne_04after": 12,
+    "lidar_velodyne_04before": 13,
+    "lidar_velodyne_05after": 14,
+    "lidar_velodyne_05before": 15,
+    "lidar_velodyne_06after": 16,
+    "lidar_velodyne_06before": 17,
+    "lidar_velodyne_07after": 18,
+    "lidar_velodyne_07before": 19,
+    "lidar_velodyne_08after": 20,
+    "lidar_velodyne_08before": 21,
+    "lidar_velodyne_09after": 22,
+    "lidar_velodyne_09before": 23,
+    "lidar_velodyne_10after": 24,
+    "lidar_velodyne_10before": 25,
+    "lidar_velodyne": 26,
+    "oxts": 27,
+}
+
+app = typer.Typer(help="Zenseact Open Dataset Downloader", no_args_is_help=True)
+
+
+def check_aria_install():
+    # assert that aria2c is installed
+    try:
+        subprocess.check_call(["aria2c", "--version"])
+    except FileNotFoundError:
+        print(
+            "aria2c is not installed. Please install aria2c to download the dataset. Using: `apt install aria2`"  # noqa
```

Comment:
would be nice to provide the user with a browser link to academic torrents, and a message stating that you don't have to install aria2 and can instead download and extract the dataset yourself with your chosen torrent client.
Bonus points if we tell the user which files to download depending on the provided flags :)

Reply:
Fixed. Dont think adding the file indices is useful.